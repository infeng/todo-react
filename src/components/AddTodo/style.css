#add-todo {
  padding: 5px 0;
}
#add-todo > input {
  border: none;
  background-color: rgba(0, 0, 0, 0);
  width: 80%;
  font-size: 1.3rem;
  vertical-align: middle;
}
#add-todo > input:focus {
  outline: none;
}
#add-todo > button {
  width: 20%;
  border: none;
  background-color: #d9534f;
  height: 30px;
  color: white;
  font-size: 1.3rem;
  vertical-align: middle;
}