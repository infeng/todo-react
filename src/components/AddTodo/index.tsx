import * as React from 'react';

import './style.css';

interface Props {
  onAddClick: (e: any) => void;
}

export default class AddTodo extends React.Component<Props, any> {
  constructor() {
    super();
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(e) {
    const node = this.refs['input'] as HTMLInputElement;
    const text: string = node.value.trim();
    if (!text) {
      return;
    }
    this.props.onAddClick(text);
    node.value = '';
  }

  render () {
    return (
      <div id="add-todo">
        <input type="text" ref="input" placeholder="What needs to be done?" />
        <button onClick={this.handleClick} >
        Add
        </button>
      </div>
    );
  }
}
