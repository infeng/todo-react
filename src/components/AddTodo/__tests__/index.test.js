import React from 'react';
import { render, mount } from 'enzyme';
import AddTodo from '..';

function createAddTodo(props) {
  return (
    <AddTodo {...props}/>
  );
}

describe('AddTodo', () => {
  it('render AddTodo correctly', () => {
    let addTodo = createAddTodo();

    let wrapper = render(addTodo);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('click', () => {
    let onAddClick = jest.fn();
    let addTodo = createAddTodo({onAddClick});

    let wrapper = mount(addTodo);
    wrapper.find('input').node.value = '233';  
    wrapper.find('button').simulate('click');
    expect(onAddClick).toBeCalledWith('233');
  });
});