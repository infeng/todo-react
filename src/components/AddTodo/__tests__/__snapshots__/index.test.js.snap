// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo render AddTodo correctly 1`] = `
<div
  id="add-todo"
>
  <input
    placeholder="What needs to be done?"
    type="text"
  />
  <button>
    Add
  </button>
</div>
`;
